test/e2e/node_problem_detector.go

```diff
@@ -193,18 +193,16 @@ var _ = framework.KubeDescribe("NodeProblemDetector", func() {
 			Expect(f.WaitForPodRunning(name)).To(Succeed())
 		})
 
-		It("should generate default node condition after started", func() {
+		It("should generate node condition and events for corresponding errors", func() {
 			By("Make sure no events are generated")
 			Consistently(func() error {
 				return verifyNoEvents(c.Events(eventNamespace), eventListOptions)
 			}, pollConsistent, pollInterval).Should(Succeed())
-			By("Make sure the node condition is false")
+			By("Make sure the default node condition is false")
 			Consistently(func() error {
 				return verifyCondition(c.Nodes(), node.Name, condition, api.ConditionFalse, defaultReason, defaultMessage)
 			}, pollConsistent, pollInterval).Should(Succeed())
-		})
 
-		It("should generate event when temporary error happens", func() {
 			num := 3
 			By(fmt.Sprintf("Inject %d temporary errors", num))
```

Comment:
If so, in the future we can add a test to update the node object with different conditions concurrently. I just use a single `TestCondition` in this test, in fact we can use a randomized condition (such as `TestCondition`+uid) to avoid the conflict. The only concern is that I'm not sure whether the node conditions will be predefined in the future, whether we could add arbitrary string in the condition field. :)